### PR TITLE
Add a helper function to present a GUI password dialog.

### PR DESCRIPTION
**Pull request by unixorn**

### CODE REVIEW

File: functions.md [hunk 1]
```diff
@@ -45,6 +47,33 @@ function mkicns() {
 }
 ```
 
+### Helper functions
```
Comment (herrbischoff) [inline functions.md]: Please capitalise correctly: Helper Functions

File: functions.md [hunk 2]
```diff
@@ -45,6 +47,33 @@ function mkicns() {
...
+
+#### Ask user for a password
```
Comment (herrbischoff) [inline functions.md]: Please capitalise and shorten correctly: Ask User for Password

File: functions.md [hunk 3]
```diff
@@ -45,6 +47,33 @@ function mkicns() {
...
+
+This function will use AppleScript to present a password entry dialog to make
+your scripts a little more user friendly.
+
+```bash
+function gui_password {
+  if [[ -z $1 ]]; then
```
Comment (herrbischoff) [inline functions.md]: Please indent with 4 spaces.

**Comment by herrbischoff** [comment]
Please excuse the late reaction. See my inline comments for changes needed to be merged.

**Comment by unixorn** [comment]
Applied the feedback.

**Comment by herrbischoff** [comment]
Thanks. But the TOC also needs to be indented by 4 spaces...

**Comment by unixorn** [comment]
Fixed the TOC indentation

**Comment by unixorn** [comment]
@herrbischoff, are there any other changes you'd like me to make?

**Comment by herrbischoff** [comment]
Thanks! Everything is alright. I was just away for some time. Thanks again for your contribution.

**Pull request merged**